src/aiida/orm/comments.py

```diff
@@ -18,7 +18,7 @@
 
 if TYPE_CHECKING:
     from aiida.orm import Node, User
-    from aiida.orm.implementation import StorageBackend
+    from aiida.orm.implementation import BackendComment, BackendNode, StorageBackend  # noqa: F401
```

Comment:
If I don't import `BackendComment` I get the following error:

```
src/aiida/orm/comments.py:63: error: Name "BackendComment" is not defined  [name-defined]
```

From this code:

https://github.com/aiidateam/aiida-core/blob/199a0276c4becfa23f64014b4f426a0c317068fe/src/aiida/orm/comments.py#L63

Reply:
The `noqa` is needed because ruff complains about an unused import. I think this is a bug in ruff, I'll file an issue.

Reply:
In seems a bit strange the type in the square of Entity is mixed, did you try `entities.Entity['BackendComment', 'CommentCollection']`? I remember my basedpyright gave something that can not use string and class for typing at same time. But doesn't matter, let's use noqa for the moment.

Reply:
Just look at the type annotation, I am wondering if it can be just have a generic type to hold both `backendComment` and `CommentCollection`?

Reply:
But is it even technically a type annotation? I don't know how the `entities.Entity` is even supposed to work, since this is somehow specifying the parent class for `Comment`.

I don't think I'd want to touch this in this PR in any case tbh.

Reply:
The needed noqa is indeed a known ruff issue: https://github.com/astral-sh/ruff/issues/9298
For now I added a following configuration to `pyproject.toml` as a workaround, per recommendation:

```
 [tool.ruff.lint.pyflakes]
 extend-generics = ["aiida.orm.entities.Entity", "aiida.orm.entities.Collection"]
```

This is unfortunate, but better than `noqa`

Reply:
Oh wow, adding that configuration actually got rid of a lot of `noqa` comments in other files, see: https://github.com/aiidateam/aiida-core/pull/6704/commits/0820342b8ee2efae8b0076cda865868b1aeb8f32